### PR TITLE
LPD-36962 don't remove the fileentry because orders would lose reference

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-36962

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-type-virtual-order-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/order/service/test/CommerceVirtualOrderItemLocalServiceTest.java [hunk 1]
```diff
@@ -275,6 +281,117 @@ public void testAddCommerceVirtualOrderItemWithSubscription()
...
+			CommerceOrderPaymentConstants.STATUS_COMPLETED,
+			CommerceOrderConstants.ORDER_STATUS_PENDING);
+
+		_commerceVirtualOrderItemChecker.checkCommerceVirtualOrderItems(
+			commerceOrder.getCommerceOrderId());
+
+		List<CommerceOrderItem> commerceOrderItems =
+			commerceOrder.getCommerceOrderItems();
+
+		List<CommerceVirtualOrderItem> userCommerceVirtualOrderItems =
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-type-virtual-order-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/order/service/test/CommerceVirtualOrderItemLocalServiceTest.java:352]: why **user**Commerce***?

File: modules/apps/commerce/commerce-product-type-virtual-order-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/order/service/test/CommerceVirtualOrderItemLocalServiceTest.java [hunk 2]
```diff
@@ -275,6 +281,117 @@ public void testAddCommerceVirtualOrderItemWithSubscription()
...
+			CommerceOrderConstants.ORDER_STATUS_PENDING,
+			commerceVirtualOrderItem.getActivationStatus());
+		Assert.assertEquals(0L, commerceVirtualOrderItem.getDuration());
+
+		if (Objects.equals(
+				commerceVirtualOrderItem.getActivationStatus(),
+				commerceOrder.getOrderStatus())) {
+
+			Assert.assertEquals(null, commerceVirtualOrderItem.getEndDate());
+		}
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-type-virtual-order-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/order/service/test/CommerceVirtualOrderItemLocalServiceTest.java:376]: for the purpose of this test are really needed all these asserts?

File: modules/apps/commerce/commerce-product-type-virtual-service/src/main/java/com/liferay/commerce/product/type/virtual/service/impl/CPDVirtualSettingFileEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -194,39 +191,9 @@ public CPDVirtualSettingFileEntry updateCPDVirtualSettingFileEntry(
...
-			}
-
-			int countCPDVirtualSettingFileEntryByFileEntryId =
-				cpdVirtualSettingFileEntryLocalService.countByFileEntryId(
-					cpdVirtualSettingFileEntryFileEntryId);
-
-			if (countCPDVirtualSettingFileEntryByFileEntryId == 0) {
-				_dlAppLocalService.deleteFileEntry(
-					cpdVirtualSettingFileEntryFileEntryId);
-			}
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-type-virtual-service/src/main/java/com/liferay/commerce/product/type/virtual/service/impl/CPDVirtualSettingFileEntryLocalServiceImpl.java:223]: in reality here we were deleting the file only if not attached to any cpdVirtualSettingFileEntry. Why we cannot extend this logic to say delete if not attached to any cpdVirtualSettingFileEntry or orderItem?

**Review by smottal — COMMENTED** [review]
More questions that changes + the sf problems 😄

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**